src/core/server/saved_objects/migrationsv2/initial_state.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as Option from 'fp-ts/Option';
+import { IndexMapping } from '../mappings';
+import { SavedObjectsMigrationVersion } from '../../../types';
+import { SavedObjectsMigrationConfigType } from '../saved_objects_config';
+import type { ISavedObjectTypeRegistry } from '../saved_objects_type_registry';
+import { InitState } from './types';
+import { excludeUnusedTypesQuery } from '../migrations/core';
+
+/**
+ * Construct the initial state for the model
+ */
+export const createInitialState = ({
+  kibanaVersion,
+  targetMappings,
+  preMigrationScript,
```

Comment:
Extracted this from `model.ts` to its own file for maintainability.

Only addition is the new `typeRegistry` parameter, and adding the `knownTypes` to the default/initial state.

Reply:
This PR is still relatively small so I guess it's ok to split files here too.
For larger efforts, it might make it harder for reviewers to figure out what has changed vs what code was moved around.